casper-finality-proofs/bin/build_circuits.rs

```diff
@@ -0,0 +1,102 @@
+use casper_finality_proofs::test_engine::wrappers::{
+    compute_shuffled_index::wrapper_mainnet::MAINNET_CIRCUIT as circuit_mainnet,
+    compute_shuffled_index::wrapper_minimal::MINIMAL_CIRCUIT as circuit_minimal,
+    wrapper_weigh_justification_and_finalization::CIRCUIT as circuit_weigh_justification_and_finalization,
+};
+use clap::Parser;
+use once_cell::sync::Lazy;
+use plonky2x::{
+    backend::circuit::{DefaultParameters, GateRegistry, HintRegistry},
+    frontend::builder::CircuitBuilder,
+};
+use std::env::args;
+use strum::{Display, EnumIter, EnumString, IntoEnumIterator};
+#[derive(Debug, Eq, Hash, PartialEq, Copy, Clone, EnumString, Display, EnumIter)]
+#[allow(non_camel_case_types)]
+enum Circuits {
+    compute_shuffled_index_mainnet,
+    compute_shuffled_index_minimal,
+    weigh_justification_and_finalization,
+    all,
+}
+
+impl Circuits {
+    fn from_str(circuit_as_str: &str) -> Option<Circuits> {
```

Comment:
EnumString already creates this function for you so this implementation is redundant. Moreover, you can provide casing in it to match the Pascal case of the enum corectly